src/javascript/app_2/App/Components/Elements/FullPageModal/full-page-modal.jsx

```diff
@@ -1,54 +1,137 @@
-import classNames from 'classnames';
-import React      from 'react';
-import PropTypes  from 'prop-types';
-import Button     from 'App/Components/Form/button.jsx';
-
-const FullPageModal = ({
-    cancel_button_text,
-    children,
-    confirm_button_text,
-    onConfirm,
-    onCancel,
-    is_visible,
-    title,
-}) => {
-    if (is_visible) {
+import classNames  from 'classnames';
+import React       from 'react';
+import PropTypes   from 'prop-types';
+import posed,
+{ PoseGroup }      from 'react-pose';
+import Button      from 'App/Components/Form/button.jsx';
+import { connect } from 'Stores/connect';
+
+const ModalWrapper = posed.div({
+    enter: {
+        y         : 0,
+        opacity   : 1,
+        delay     : 300,
+        transition: {
+            y      : { type: 'spring', stiffness: 1000, damping: 15 },
+            default: { duration: 250 },
+        },
+    },
+    exit: {
+        y         : 50,
+        opacity   : 0,
+        transition: { duration: 250 },
+    },
+});
+
+const ModalBackground = posed.div({
+    enter: { opacity: 1 },
+    exit : { opacity: 0 },
+});
+
+class FullPageModal extends React.Component {
+    componentDidMount() {
+        if (this.props.is_visible) {
+            this.props.showAppBlur();
+        }
+    }
+
+    componentDidUpdate() {
```

Comment:
is this necessary since adding the blur effect is done onMount?

Reply:
yes, since there are instances where the prop is updated and we need to re-toggle the `blur` effect. It's mainly for the `is_virtual_account` prop check when client switches account from the accounts switcher and not the `modal`.